components/StorageFileList.vue

```diff
@@ -0,0 +1,37 @@
+<script setup lang="ts">
+import { FileObject } from "@/types/FileObject";
+
+const client = useSupabaseClient();
+
+const fileList = ref<FileObject[] | null>([]);
+
+const listAllFiles = async () => {
+  const { data, error } = await client.storage.from("files").list("public", {
+    limit: 100,
+    offset: 0,
+    sortBy: { column: "name", order: "asc" },
+  });
+  if (error) throw error;
+  data?.forEach((file) => {
+    fileList.value?.push(file);
+  });
+};
+
+listAllFiles();
```

Comment:
I added onMounted hook cuz I think I don't need to call the api on the server side, since the list it is not on the homepage and the list could change when user adds new files - then I need api call on the client side.